mewarpx/mewarpx/diags_store/particle_diagnostic.py

```diff
@@ -0,0 +1,154 @@
+from mewarpx.mwxrun import mwxrun
+
+logger = logging.getLogger(__name__)
+
+class ParticleDiagnostic(WarpXDiagnostic):
+    """Output particle diagnostics every diagnostic period and produce
+        plots from the diagnostics
+
+    Contains:
+        add_particle_diag (function): Function to add the particle diagnostic to
+        the mwxrun.simulation
+
+        post_processing: plot particle diagnostic data, not yet implemented
+    """
+
+    def __init__(self, diag_steps, name=None, species=None,
+                data_list=None, write_dir=".",
+                post_process=False, plot_data_list=None,
+                plot_species=None, **kwargs):
+
+        """
+        Arguments:
+            Initializes the picmi.ParticleDiagnostic and adds the diagnostic to
+            the simulation
+
+            diag_steps (int): Number of steps between each diagnostic output
+
+            name (str): name of the diag output folder, defaults to diag
+
+            species (mepicmi.Species): species in the simulation, if None then
+            uses all particles in the simulation
+
+            data_list (list str): list of attributes to be outputted by he
+            diagnostic, default uses ["position", "momentum", "weighting"]
+
+            write_dir (str): where to write particle diagnostic data
+
+            post_process (bool): generate plots for each diagnostic data
+            directory produced
+
+            plot_data_list (list str): list of data to be plotted for each
+            diagnostic step ("particle_position_x", "particle_position_y",
+            "particle_position_z", "particle_momentum_x", "particle_momentum_y",
+            "particle_momentum_z")
+
+            plot_species (list str): list of species names to be plotted,
+            defaults to all species if not specified. Name variable in
+            mepicmi.Species must be set for each species in the simulation
+
+        """
+        self.name = name
+        self.species = species
+        self.data_list = data_list
+        self.write_dir = write_dir
+
+        self.post_process = post_process
+        self.plot_data_list = plot_data_list
+        self.plot_species = plot_species
+
+        if self.name is None:
+            self.name= "diag"
+        if self.species is None:
+            self.species = mwxrun.simulation.species
+        if self.data_list is None:
+            self.data_list = ["position", "momentum", "weighting"]
+        if self.plot_species is None:
+            self.plot_species = []
+            for specimen in self.species:
+                self.plot_species.append(specimen.name)
+
+        super(ParticleDiagnostic, self).__init__(diag_steps=diag_steps, **kwargs)
+        self.add_particle_diag()
+        if self.post_processing:
+            callbacks.installafterstep(self.check_for_end_of_sim)
+
+        if self.post_process:
+            if self.plot_data_list is None:
+                raise UserWarning("Warning: post_process is True,"
```

Comment:
```suggestion
                warnings.warn("Warning: post_process is True,"
```

Reply:
Note that `raise` would make it act like an exception, not a warning. If you do want an exception, then make it a `ValueError`.